modules/arrow/src/schema/convert-arrow-schema.ts

```diff
@@ -142,11 +195,27 @@ export function serializeArrowType(arrowType: ArrowDataType): DataType {
       return 'timestamp-microsecond';
     case TimestampNanosecond:
       return 'timestamp-nanosecond';
-    // case Interval: return 'interval';
+    case Interval:
+      const intervalUnit = (arrowType as Interval).unit;
+      switch (intervalUnit) {
+        case IntervalUnit.DAY_TIME:
+          return 'interval-daytime';
+        case IntervalUnit.YEAR_MONTH:
+          return 'interval-yearmonth';
+        default:
+          return 'interval-daytime';
+      }
     case IntervalDayTime:
       return 'interval-daytime';
     case IntervalYearMonth:
       return 'interval-yearmonth';
+    case List:
+      const listType = arrowType as List;
+      const listField = listType.valueField;
+      return {
+        type: 'list',
+        children: [serializeArrowField(listField)]
+      };
```

Comment:
Need to add `List` class type when serialization, e.g. for a polygon field, the field type could be like `List<List<FixedSizeList[2]<Float64>>>`